internal/models/verification.go

```diff
@@ -1,39 +1,98 @@
+	BeneficiaryID         string               `bson:"beneficiaryId" json:"beneficiaryId"`     // required
+	RegistryCenterCheck   interface{}          `json:"registryCenterCheck,omitempty"`
+	AddressVerification   interface{}          `json:"addressVerification,omitempty"`
+	QuestionnaireAnswers  interface{}          `json:"questionnaireAnswers,omitempty"`
+	AdditionalSteps       map[string]string    `json:"additionalSteps,omitempty"`
+	UtilityData           []string             `json:"utilityData,omitempty"`
+	AdditionalStepPdfUrls map[string]string    `json:"additionalStepPdfUrls,omitempty"`
+	AML                   []AMLCheck           `bson:"AML" json:"AML,omitempty"`
+	LID                   []LID                `bson:"LID" json:"LID,omitempty"`
+	ExternalRef           string               `bson:"externalRef" json:"externalRef,omitempty"`
+	ManualAddress         string               `bson:"manualAddress" json:"manualAddress,omitempty"`
+	ManualAddressMatch    *bool                `bson:"manualAddressMatch" json:"manualAddressMatch,omitempty"`
+	ExpirationStatus      *ExpirationThreshold `bson:"expirationStatus,omitempty" json:"expirationStatus,omitempty"`
+}
+
+// implements slog.LogValuer to control how Verification is logged
+func (v Verification) LogValue() slog.Value {
+	// Create a copy without sensitive data
+	sanitized := &Verification{
+		Final:            v.Final,
+		Platform:         v.Platform,
+		Status:           v.Status,
+		IdenfyRef:        v.IdenfyRef,
+		ClientID:         v.ClientID,
+		StartTime:        v.StartTime,
+		FinishTime:       v.FinishTime,
+		ExpirationStatus: v.ExpirationStatus,
+	}
+
+	// Convert to a map for logging
+	return slog.GroupValue(
+		slog.Any("final", sanitized.Final),
+		slog.String("platform", string(sanitized.Platform)),
+		slog.Any("status", sanitized.Status),
+		slog.String("idenfyRef", sanitized.IdenfyRef),
+		slog.String("clientId", sanitized.ClientID),
+		slog.Int64("startTime", sanitized.StartTime),
+		slog.Int64("finishTime", sanitized.FinishTime),
+		slog.Any("expirationStatus", sanitized.ExpirationStatus),
+	)
+}
+
+// ToOutcome converts a Verification to a VerificationOutcome based on the service configuration
+func (v Verification) ToOutcome(config config.Verification) *VerificationOutcome {
+	outcome := OutcomeRejected
+	// First check if Overall status exists
+	if v.Status.Overall != nil {
+		// Then evaluate if it's either:
+		// 1. Overall status is Approved, or
+		// 2. Overall status is Suspected AND config allows suspicious cases to be approved
+		if *v.Status.Overall == OverallApproved ||
+			(*v.Status.Overall == OverallSuspected && config.SuspiciousVerificationOutcome == "APPROVED") {
+
+			// If either condition is met, set outcome to Approved
+			outcome = OutcomeApproved
+
+			// Finally check if document is expired - this overrides the Approved outcome based on config
+			if v.ExpirationStatus != nil && *v.ExpirationStatus == DocumentExpired {
```

Comment:
if i could suggest simpler structure to avoid the nesting block
```go
verOutcome := &VerificationOutcome{
		Final:               v.Final,
		ClientID:            v.ClientID,
		IdenfyRef:           v.IdenfyRef,
		ExpirationThreshold: v.ExpirationStatus,
		Outcome:             OutcomeRejected,
	}

if v.Status.Overall == nil {
  return verOutcome
}

// continue with other conditions to override verOutcome.outcome and return 
```